### ISSUE TITLE
Implementing Finetuning by RLHF and explores COT

### ISSUE THREAD

**Issue by WazupSteve** [huggingface/huggingface-llama-recipes#56]
Hey @ariG23498 , Thanks for the open call! I’m adding a comment here to ask if this can be part of the cookbook.
A recipe that explores fine-tuning LLaMA using reinforcement learning. This involves setting up reward functions and creating pipelines for RL-based fine-tuning. A different LLaMa model can be used as a teacher model to guide training for smaller models or set-up an environment for the model to do CoT like o1.

**Comment by ariG23498** [comment]
Hey @WazupSteve

This is a really nice project on its own! Unfortunately we are not looking for something like this for our repository. The llama-recipes repo was built in order to house code snippets that are simple for any beginner to start working with the Llama family of models.

This example is a very thorough project which should have its own place. Closing this issue, please feel free to suggest something that aligns with the type of code that already exists in the repository! 🤗
